### ISSUE TITLE
Latest version of PTVSD throws errors when debugging

### ISSUE THREAD

**Issue by DonJayamanne** [Microsoft/ptvsd#657]
## Environment data

- PTVSD version: Master
- OS and version: Mac
- Python version (& distribution if applicable, e.g. Anaconda): N/A
- Using VS Code or Visual Studio:VSC

## Actual behavior

* Errors displayed in terminal

## Expected  behavior

* Error should not be displayed

## Steps to reproduce:
1. Debug a simple `hello world` python file
2. The following error is displayed in the terminal
```
Traceback (most recent call last):
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/pydevd_file_utils.py", line 449, in get_abs_path_real_path_and_base_from_file
    return NORM_PATHS_AND_BASE_CONTAINER[f]
KeyError: '/Users/donjayamanne/Desktop/Development/PythonStuff/vscodePythonTesting/dbg/venv/lib/python36.zip'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/pydevd_file_utils.py", line 202, in _NormPaths
    return NORM_PATHS_CONTAINER[filename]
KeyError: '/Users/donjayamanne/Desktop/Development/PythonStuff/vscodePythonTesting/dbg/venv/lib/python36.zip'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command.py", line 758, in process_net_command
    pydevd_utils.set_project_roots(text.split(u'\t'))
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_utils.py", line 225, in set_project_roots
    project_roots = _set_roots(project_roots, _PROJECT_ROOTS_CACHE)
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_utils.py", line 172, in _set_roots
    new_roots.append(_normpath(root))
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_utils.py", line 19, in _normpath
    return pydevd_file_utils.get_abs_path_real_path_and_base_from_file(filename)[0]
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/pydevd_file_utils.py", line 451, in get_abs_path_real_path_and_base_from_file
    abs_path, real_path = _NormPaths(f)
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/pydevd_file_utils.py", line 204, in _NormPaths
    abs_path = _NormPath(filename, os.path.abspath)
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/_vendored/pydevd/pydevd_file_utils.py", line 220, in _NormPath
    if r[ind] == "!":
IndexError: string index out of range
```

## Note:
* Either I'm doing something completely wrong
* Or our tests (CI) are seriously faulty

**Comment by karthiknadig** [comment]
This was introduced by https://github.com/Microsoft/ptvsd/commit/e9ed23fe75b4aa596dfc8cb80361f2a07feb6c8d. Also same as #656 

/cc @fabioz

**Comment by DonJayamanne** [comment]
So, a very simple and critical bug, managed to get through all tests!!
I guess this test would have caught this issue https://github.com/Microsoft/ptvsd/issues/570

**Comment by fabioz** [comment]
Investigating/working on a fix.

**Comment by fabioz** [comment]
Just provided a pull request for this issue at: https://github.com/Microsoft/ptvsd/pull/660 (it seems that the Travis build got stuck before doing anything, so, it needs to be restarted).

**Comment by karthiknadig** [comment]
Fixed via #660
